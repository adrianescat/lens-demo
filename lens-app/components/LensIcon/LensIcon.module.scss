.lensIcon {
  width: 20px;
}