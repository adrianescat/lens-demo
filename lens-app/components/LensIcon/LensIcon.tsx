import React, { ReactElement } from 'react'
import { faFileMedical, faFlask, faFilePrescription } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import classnames from 'classnames';
import styles from './LensIcon.module.scss';
import { LensType } from '../../types/lens';

const iconMap = {
  [LensType.HISTORY]: faFileMedical,
  [LensType.METAB_LAB_REPORT]: faFlask,
  [LensType.RX_PRESC_REQ]: faFilePrescription,
}


interface Props {
  label: string;
  className?: string;
  title: string;
}

export default function LensIcon({ label, className, title }: Props): ReactElement {
  return (
    <div className={classnames(styles.lensIcon, className)} title={title} aria-label={title}>
      <FontAwesomeIcon className={styles.flagIcon} icon={iconMap[label]} />
    </div>
  )
}
