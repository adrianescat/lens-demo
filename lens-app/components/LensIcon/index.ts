import LensIcon from './LensIcon';

export default LensIcon;