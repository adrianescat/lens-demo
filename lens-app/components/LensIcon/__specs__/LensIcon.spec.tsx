import React from 'react';
import { shallow } from 'enzyme';
import LensIcon, { Props } from '../LensIcon';

let wrapper: any;
let props: Props;

beforeEach(() => {
  props = { label: 'label', title: 'title' };
  wrapper = shallow(<LensIcon {...props} />);
});

test('should match snapshots', () => {
  expect(wrapper).toMatchSnapshot();
});