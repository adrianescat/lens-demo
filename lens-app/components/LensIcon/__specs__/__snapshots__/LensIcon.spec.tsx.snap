// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<div
  aria-label="title"
  className="lensIcon"
  title="title"
>
  <FontAwesomeIcon
    border={false}
    className="flagIcon"
    fixedWidth={false}
    flip={null}
    icon={null}
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</div>
`;
